s3tests_boto3/functional/test_iam.py

```diff
@@ -934,3 +935,619 @@ def test_verify_allow_iam_actions():
     response = iam_client_alt.delete_user_policy(PolicyName='AllowAccessPolicy',
                                           UserName=get_alt_user_id())
     eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow Bucket Actions with delimiter condition statement')
+@attr(assertion='succeeds')
+@attr('user-policy')
+@attr('test_of_iam')
+def test_allow_delimiter_condition_in_user_policy():
+    client = get_iam_client()
+    s3_client_iam = get_iam_s3client()
+    s3_client_alt = get_alt_client()
+    bucket = get_new_bucket(client=s3_client_iam)
+    s3_client_iam.put_object(Bucket=bucket, Key='test/foo1', Body='bar')
+    s3_client_iam.put_object(Bucket=bucket, Key='test/secondary/foo2', Body='bar')
```

Comment:
You might keep delimiter as any usual delimiter like /. Keeping secondary should not list test/foo1 and same is the case for   test/secondary/foo2